src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, NgZone, OnDestroy } from '@angular/core';
+import { uniqueIdFactory } from '../id-generator/id-generator.service';
+
+// export type AriaLivePoliteness = 'off' | 'polite' | 'assertive';
+export enum AriaLivePoliteness {
+  off = 'off',
+  polite = 'polite',
+  assertive = 'assertive',
+}
+
+@Injectable()
+export class AriaLiveService implements OnDestroy {
+  private _TICK: number = 100;
+  private _ariaLiveElement: HTMLElement;
+  private _document: Document;
+  private _previousTimeout?: number;
+
+  private _id: string = `clr-aria-live-element-${uniqueIdFactory()}`;
+
+  constructor(private _ngZone: NgZone, @Inject(DOCUMENT) _document: any) {
+    this._document = _document;
+    this._ariaLiveElement = this._createLiveElement();
+  }
+
+  private _createLiveElement(): HTMLElement {
+    const ariaLiveElement = this._document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.getId());
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
+    ariaLiveElement.setAttribute('aria-live', AriaLivePoliteness.polite);
+
+    this._document.body.appendChild(ariaLiveElement);
+
+    return ariaLiveElement;
+  }
+
+  public announce(message: string, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): Promise<void> {
+    this.clear();
+    clearTimeout(this._previousTimeout);
+
+    this._ariaLiveElement.setAttribute('aria-live', politeness);
+
+    // This 100ms timeout is necessary for some browser + screen-reader combinations:
+    // - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
+    // - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
+    //   second time without clearing and then using a non-zero delay.
+    // (using JAWS 17 at time of this writing).
+    return this._ngZone.runOutsideAngular(() => {
+      return new Promise(resolve => {
```

Comment:
My use here is this 
```ts
ariaLiveService.announce('Some update text')
  .then(() => { this.triggerSomeMethodd });
```

You could know exactly when the event is pushed and could take some action after it.

Reply:
But we don't have a use case for it currently though. I don't see any place where we are using this promise. We shouldn't just add something when we don't know for sure it would be useful or not in the future. 

And if we really need to listen to this event, why Promise...? Why not Observable?